Adds method RigidBody::add_joint

amcastro-tri opened the pull request:
This PR simply adds an API to add joints to the RBT. 
Method `AddFloatingJoint` was the only API available to add joints to RBT and got removed by #3358. PR #3245 depended on it and I needed a clean solution. 
Therefore I implemented an API to add joints. Since joints cannot be added without having a body attaching to it, the method belongs to the `RigidBody` (think of the joint as adding DOF's).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/3424)
<!-- Reviewable:end -->

amcastro-tri commented:
+@sherm1 for platform please.
+@liangfok for feature please since you removed `AddFloatingJoint` in #3358.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRVnYMwjHznjwP3kduR:bfgippu)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Except for a change to a comment, <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> - PTAL.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/plants/RigidBody.h, line 62 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVuVTTv_7h0gF8ux8E:-KRVuVTTv_7h0gF8ux8F:b-ucm4ce) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L62)):*
> ```objc
>   /**
>    * Adds degrees of freedom to this body by connecting it to @p parent with
>    * @p joint. The body takes ownership of the joint.
> ```

Please add something like: `Note that this is specifically a tree joint (mobilizer) and that by "parent" we mean a body that is closer to "world" in the tree topology.`

(Optional) You could say "specifies the mobility of this body relative to its parent body" in addition to "adds degrees of freedom".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRVw9bR5J6aJ8ZfvlqX:bjo1s1r)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Completed first pass. PTAL.

---

Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/systems/plants/RigidBody.h, line 61 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVv9CPT8VBRIzMJUDC:-KRVv9CPT8VBRIzMJUDD:b-bb23iy) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L61)):*
> ```objc
> 
>   /**
>    * Adds degrees of freedom to this body by connecting it to @p parent with
> ```

I believe @sherm1 prefers "a mobilizer" instead of "degrees of freedom".

---

*[drake/systems/plants/RigidBody.h, line 62 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVuVTTv_7h0gF8ux8E:-KRVy4ZKvuhVyrDr34bu:b-c3d9yx) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L62)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> Please add something like: `Note that this is specifically a tree joint (mobilizer) and that by "parent" we mean a body that is closer to "world" in the tree topology.`
> 
> (Optional) You could say "specifies the mobility of this body relative to its parent body" in addition to "adds degrees of freedom".
</details>
How about let's replace "parent" with "inboard" and "joint" with "mobilizer" per https://github.com/RobotLocomotion/drake/issues/3320?

---

*[drake/systems/plants/RigidBody.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvQ8pLPorZQ4m_Fm-:-KRVvQ8qj_vSboTd9ZyV:b-ah80jp) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L64)):*
> ```objc
>    * @p joint. The body takes ownership of the joint.
>    *
>    * @param parent The RigidBody this body gets connected to.
> ```

BTW. Add parameter direction: `@param[in]`.

---

*[drake/systems/plants/RigidBody.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvuB1Uu1CNmcZ9W4z:-KRVvuB1Uu1CNmcZ9W5-:b-kdls0n) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L64)):*
> ```objc
>    * @p joint. The body takes ownership of the joint.
>    *
>    * @param parent The RigidBody this body gets connected to.
> ```

Mention that the pointer is copied and stored meaning its lifetime must exceed the lifetime of this `RigidBody`.

---

*[drake/systems/plants/RigidBody.h, line 65 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvli9u8jzM65MGuCu:-KRVvli9u8jzM65MGuCv:bubnsxb) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L65)):*
> ```objc
>    *
>    * @param parent The RigidBody this body gets connected to.
>    * @param joint The DrakeJoint connecting this body to @p parent and adding
> ```

BTW. Add parameter direction: `@param[in]`.

---

*[drake/systems/plants/RigidBody.h, line 69 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVwKzpx1L7OVw_sOhQ:-KRVwKzpx1L7OVw_sOhR:byxe6gp) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L69)):*
> ```objc
>    * @returns A pointer to the joint just added to the body.
>    */
>   template<typename JointType>
> ```

Why use templates here instead of `DrakeJoint` as the parameter type? That would allow us to move the implementation into the `.cc` file.

---

*[drake/systems/plants/RigidBody.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVxEUb1Pn2-fMTQE6N:-KRVxEUb1Pn2-fMTQE6O:b-tdslzu) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L73)):*
> ```objc
>     JointType* joint_ptr = joint.get();
>     set_parent(parent);
>     setJoint(move(joint));
> ```

Given this method, can we deprecate `setJoint()`?

---

*[drake/systems/plants/RigidBody.h, line 74 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVwz2JV2wSQVizCvn-:-KRVwz2JV2wSQVizCvn0:b-6rv2dy) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L74)):*
> ```objc
>     set_parent(parent);
>     setJoint(move(joint));
>     return joint_ptr;
> ```

BTW, why not make this line:

```
return joint_.get();
```

?

That would enable you to delete the line containing:

```
JointType* joint_ptr = joint.get();
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRVxiN3jOdG5eSv5KdL:bvu844y)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 1 of 2 files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/systems/plants/RigidBody.h, line 61 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVv9CPT8VBRIzMJUDC:-KRVzDuqHjzsuZ9JrddT:bazahu9) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L61)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> I believe @sherm1 prefers "a mobilizer" instead of "degrees of freedom".
</details>
We will add those terms to the dynamics engine 2.0. I don't think those terms belong at all here in RBT (which only knows about "joints"). I am planning to discuss the new plan for Dynamics 2.0 in the Friday meeting.

---

*[drake/systems/plants/RigidBody.h, line 62 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVuVTTv_7h0gF8ux8E:-KRVzkx7L2QzbDm2003Z:b-6d8xy8) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L62)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> How about let's replace "parent" with "inboard" and "joint" with "mobilizer" per https://github.com/RobotLocomotion/drake/issues/3320?
</details>
I agree those are the proper terms to be used in Dynamics 2.0. I won't try to make substantial changes or introduce new terminology into the RBT in this PR (I think that only adds confusion to RBT users). I did however add a note about the parent being closer to the world.

---

*[drake/systems/plants/RigidBody.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvQ8pLPorZQ4m_Fm-:-KRW-CB4jbzUOfu3Os9i:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L64)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW. Add parameter direction: `@param[in]`.
</details>
Done.

---

*[drake/systems/plants/RigidBody.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvuB1Uu1CNmcZ9W4z:-KRW-SuGWcPhTIBFBXIg:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L64)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Mention that the pointer is copied and stored meaning its lifetime must exceed the lifetime of this `RigidBody`.
</details>
Done.

---

*[drake/systems/plants/RigidBody.h, line 65 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVvli9u8jzM65MGuCu:-KRW-UhCLkP21IwX0x6o:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L65)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW. Add parameter direction: `@param[in]`.
</details>
Done.

---

*[drake/systems/plants/RigidBody.h, line 69 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVwKzpx1L7OVw_sOhQ:-KRW-aSV6JGriZQVoD_G:b-9dquyr) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L69)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Why use templates here instead of `DrakeJoint` as the parameter type? That would allow us to move the implementation into the `.cc` file.
</details>
Because the argument is an `std::unique_ptr`. That'd be possible if the argument was just a raw pointer. It's a pretty small method though so I'm not worried about compilation time here, no heavy computations.

---

*[drake/systems/plants/RigidBody.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVxEUb1Pn2-fMTQE6N:-KRW-t9JikrAsw-G5dgK:bvxtp31) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L73)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Given this method, can we deprecate `setJoint()`?
</details>
They do different things so I don't think so. There are parser methods still using it.

---

*[drake/systems/plants/RigidBody.h, line 74 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVwz2JV2wSQVizCvn-:-KRW030T2hTQ5tkgoIQM:b18ehml) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L74)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW, why not make this line:
> 
> ```
> return joint_.get();
> ```
> 
> ?
> 
> That would enable you to delete the line containing:
> 
> ```
> JointType* joint_ptr = joint.get();
> ```
</details>
nice catch. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Done with second pass. PTAL.

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/systems/plants/RigidBody.h, line 61 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVv9CPT8VBRIzMJUDC:-KRWEOD3eMs3ArpBf-tF:b2nhktp) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L61)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> We will add those terms to the dynamics engine 2.0. I don't think those terms belong at all here in RBT (which only knows about "joints"). I am planning to discuss the new plan for Dynamics 2.0 in the Friday meeting.
</details>
OK.

---

*[drake/systems/plants/RigidBody.h, line 62 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVuVTTv_7h0gF8ux8E:-KRWDyDOq7sbvj9Q-HF6:b2nhktp) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L62)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> I agree those are the proper terms to be used in Dynamics 2.0. I won't try to make substantial changes or introduce new terminology into the RBT in this PR (I think that only adds confusion to RBT users). I did however add a note about the parent being closer to the world.
</details>
OK.

---

*[drake/systems/plants/RigidBody.h, line 69 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVwKzpx1L7OVw_sOhQ:-KRWEMXO9UMHTXbrOvOX:b2nhktp) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L69)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> Because the argument is an `std::unique_ptr`. That'd be possible if the argument was just a raw pointer. It's a pretty small method though so I'm not worried about compilation time here, no heavy computations.
</details>
OK.

---

*[drake/systems/plants/RigidBody.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVxEUb1Pn2-fMTQE6N:-KRWD_EXFVUOBxO1I0ad:bjo0nfy) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L73)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> They do different things so I don't think so. There are parser methods still using it.
</details>
Can you add the following TODO comment to method `setJoint()` as a reminder to myself?

```
TODO(liang.fok): Deprecate this method in favor of add_joint(). This requires, among potentially other things, updating the parsers.
```

---

*[drake/systems/plants/RigidBody.h, line 61 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRWE37rirTHpdPzCrzk:-KRWE37s5ACT8V05cOXV:b-rnmtng) ([raw file](https://github.com/robotlocomotion/drake/blob/c5a3a2b199930e00434f2aeb51b90227023a2c9c/drake/systems/plants/RigidBody.h#L61)):*
> ```objc
> 
>   /**
>    * Adds degrees of freedom to this body by connecting it to @p parent with
> ```

Please document what happens if this method is called on a `RigidBody` that already has a joint. I believe the old joint is lost right? Maybe there should be an assert statement preventing this from happening?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRWEYc7PG1nwKEbiZt2:b1dl0dt)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/systems/plants/RigidBody.h, line 73 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRVxEUb1Pn2-fMTQE6N:-KRWiz-fwNFdVGJ_g5Se:b80qo8t) ([raw file](https://github.com/robotlocomotion/drake/blob/d7974c13dd736f6a87d123e65712a059a4e06f99/drake/systems/plants/RigidBody.h#L73)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Can you add the following TODO comment to method `setJoint()` as a reminder to myself?
> 
> ```
> TODO(liang.fok): Deprecate this method in favor of add_joint(). This requires, among potentially other things, updating the parsers.
> ```
</details>
Done, thanks for volunteering. 

---

*[drake/systems/plants/RigidBody.h, line 61 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRWE37rirTHpdPzCrzk:-KRWiHiZ3jJto2Q4keOU:b-nry6rt) ([raw file](https://github.com/robotlocomotion/drake/blob/c5a3a2b199930e00434f2aeb51b90227023a2c9c/drake/systems/plants/RigidBody.h#L61)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Please document what happens if this method is called on a `RigidBody` that already has a joint. I believe the old joint is lost right? Maybe there should be an assert statement preventing this from happening?
</details>
Correct. Added a check + abort and updated doc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Done with third pas. PTAL. CI failure is real; one of my comments below should address it.

---

Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[drake/systems/plants/RigidBody.cpp, line 39 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYg37tfbYC13x3qwuR:-KRYg37tfbYC13x3qwuS:biiarc8) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.cpp#L39)):*
> ```cpp
> 
> // TODO(liang.fok): Deprecate this method in favor of add_joint().
> // This requires, among potentially other things, updating the parsers.
> ```

Can you copy / paste this message into the method declaration in `RigidBody.h`? Thanks.

---

*[drake/systems/plants/RigidBody.h, line 66 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYgJPP2QsfTtEuW_lV:-KRYgJPP2QsfTtEuW_lW:beknyqh) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.h#L66)):*
> ```objc
>    *
>    * This method aborts with an error message if the user attempts to assign a
>    * joint to a body which already has one.
> ```

BTW: nit: replace "which" with "that".

---

*[drake/systems/plants/RigidBody.h, line 81 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYewwNdm9Uh0KI_q0p:-KRYewwNdm9Uh0KI_q0q:b-ojliyk) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.h#L81)):*
> ```objc
>   template<typename JointType>
>   JointType* add_joint(RigidBody* parent, std::unique_ptr<JointType> joint) {
>     if (!joint_) {
> ```

Should this be:

```
if (joint_) {
```

?

Or, to be even more clear, maybe it should be:

```
if (joint_.get() != nullptr) {
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRYga-nZJZlLqR5KcJl:bg560os)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/systems/plants/RigidBody.h, line 66 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYgJPP2QsfTtEuW_lV:-KRZGaAkAhCgnKMfvo88:bo3vl6y) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.h#L66)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW: nit: replace "which" with "that".
</details>
Thanks, done.

---

*[drake/systems/plants/RigidBody.h, line 81 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYewwNdm9Uh0KI_q0p:-KRZMqpjntC95HfnVwYg:b-4viqd) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.h#L81)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Should this be:
> 
> ```
> if (joint_) {
> ```
> 
> ?
> 
> Or, to be even more clear, maybe it should be:
> 
> ```
> if (joint_.get() != nullptr) {
> ```
</details>
Yes, the condition was wrong. It should be fixed now.

---

*[drake/systems/plants/RigidBody.cpp, line 39 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-KRYg37tfbYC13x3qwuR:-KRZGUmls5y1-wQDaJRR:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/6c3b3d966465e316cfa7f31e376f4354c8445bbf/drake/systems/plants/RigidBody.cpp#L39)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Can you copy / paste this message into the method declaration in `RigidBody.h`? Thanks.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3424#-:-KRZZgAkXAgi-ONoG0CG:bnfp4nl)*
<!-- Sent from Reviewable.io -->
